File: Data Structures & Algorithms/longest-consecutive-sequence/submission-10.cpp
#include <cmath>
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int c = 0;
        unordered_set<int> s(nums.begin(), nums.end());
        for (int r = 0; r <nums.size(); r++) {
            int n = nums[r], ch = 1;
            while(s.count(n-1)) {
                ch++;
                n--;
            }
            c = max(c,ch);
        }
        return c;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-10.py
class Solution:
    def trap(self, height: List[int]) -> int:
        if not height:
            return 0
        l, r, m = 0, len(height)-1, 0
        lm, rm = height[l], height[r]
        while l <r:
            if height[l] < height[r]:
                l+=1
                lm = max(height[l], lm)
                m+= lm - height[l]
            else:
                r-=1
                rm = max(height[r], rm)
                m+= rm - height[r]
        return m